### PR TITLE
Add an endpoint to fetch users by graffiti

### PR DESCRIPTION
**Pull request by mat-if**
## Testing Plan

Unit test, manual test

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/users/users.controller.ts [hunk 1]
```diff
@@ -75,6 +75,25 @@ export class UsersController {
...
+  @Get('/graffiti/:graffiti')
+  async findByGraffiti(
+    @Param(
+      'graffiti',
+      new ValidationPipe({
+        errorHttpStatusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+      }),
+    )
+    graffiti: string,
+  ): Promise<SerializedUser> {
```
Comment (rohanjadvani) [inline src/users/users.controller.ts:89]: nit:
```suggestion
  ): Promise<SerializedUserWithRank> {
```

I just realized the `find` method above should probably return this type too.

File: src/users/users.controller.ts [hunk 2]
```diff
@@ -75,6 +75,25 @@ export class UsersController {
     );
   }
 
+  @ApiOperation({ summary: 'Gets a specific User' })
+  @ApiParam({ description: 'Unique User graffiti', name: 'graffiti' })
+  @Get('/graffiti/:graffiti')
```
[the remark this answers is not in the document]
Comment (mat-if) [reply src/users/users.controller.ts:80]: Good call. `/graffiti/:graffiti` felt pretty awkward, I like that a lot more.

File: src/users/users.controller.spec.ts [hunk 3]
```diff
@@ -64,6 +65,70 @@ describe('UsersController', () => {
...
+          const user = await prisma.user.create({
+            data: {
+              email: faker.internet.email(),
+              graffiti: graffiti,
+              country_code: faker.address.countryCode(),
+            },
+          });
+          const { body } = await request(app.getHttpServer())
+            .get(`/users/find`)
+            .query({ graffiti: graffiti, with_rank: true })
```
Comment (rohanjadvani) [inline src/users/users.controller.spec.ts:82]:
```suggestion
            .query({ graffiti, with_rank: true })
```

File: src/users/users.controller.spec.ts [hunk 4]
```diff
@@ -64,6 +65,70 @@ describe('UsersController', () => {
...
 
+  describe('GET /users/find', () => {
+    describe('with rank requested', () => {
+      describe('with a valid graffiti', () => {
+        it('returns the user', async () => {
+          const graffiti = uuid();
+          const user = await prisma.user.create({
+            data: {
+              email: faker.internet.email(),
+              graffiti: graffiti,
```
Comment (rohanjadvani) [inline src/users/users.controller.spec.ts:76]:
```suggestion
              graffiti,
```

File: src/users/users.controller.spec.ts [hunk 5]
```diff
@@ -64,6 +65,70 @@ describe('UsersController', () => {
...
+    });
+
+    describe('with rank not requested', () => {
+      describe('with a valid graffiti', () => {
+        it('returns the user', async () => {
+          const graffiti = uuid();
+          const user = await prisma.user.create({
+            data: {
+              email: faker.internet.email(),
+              graffiti: graffiti,
```
Comment (rohanjadvani) [inline src/users/users.controller.spec.ts:103]:
```suggestion
              graffiti,
```

File: src/users/users.controller.spec.ts [hunk 6]
```diff
@@ -64,6 +65,70 @@ describe('UsersController', () => {
...
+              country_code: faker.address.countryCode(),
+            },
+          });
+          const { body } = await request(app.getHttpServer())
+            .get(`/users/find`)
+            .query({ graffiti: graffiti })
```
Comment (rohanjadvani) [inline src/users/users.controller.spec.ts:109]:
```suggestion
            .query({ graffiti })
```

File: src/users/users.controller.spec.ts [hunk 7]
```diff
@@ -64,6 +65,70 @@ describe('UsersController', () => {
...
+            created_at: user.created_at.toISOString(),
+          });
+        });
+      });
+    });
+
+    describe('with a missing graffiti', () => {
+      it('returns a 404', async () => {
+        await request(app.getHttpServer())
+          .get('/users/graffiti/12345')
```
Comment (rohanjadvani) [inline src/users/users.controller.spec.ts:126]: Should this use the new find endpoint?

**Review by rohanjadvani — COMMENTED** [review]
A couple nits, but have one clarifying question regarding a test

File: src/users/users.controller.spec.ts (hunk 7 above)
Comment (mat-if) [reply src/users/users.controller.spec.ts]: YEP good catch, fixed

**Review by rohanjadvani — APPROVED** [review]

**Pull request merged**